homeassistant/components/entur_public_transport/sensor.py

```diff
@@ -100,12 +106,19 @@ async def async_setup_platform(
     stop_ids = config[CONF_STOP_IDS]
     omit_non_boarding = config[CONF_OMIT_NON_BOARDING]
     number_of_departures = config[CONF_NUMBER_OF_DEPARTURES]
+    # Prepare uuid for API request. Needs to be encoded for hashing:
+    url = (get_url(hass) + str(randint(100, 999))).encode(
+        encoding="UTF-8", errors="strict"
+    )
+    uuid = (
```

Comment:
There is no reason to make this based on any user input. You can just generate a new ULID when the integration is set up and which changes every restart.

Reply:
That's of course also an option, @balloob. So you're suggesting I nuke the configuration-option altogether? I'd still compute the uuid in the above way which feels cleaner than purely random and the server-side would probably also appreciate a stable id -- I could easily see that frequent restarts e.g. during development would otherwise get your instance rate-limited for a while (happened to me with another plugin that got hit by rate-limiting because I was restarting frequently).

Reply:
There is no reason to share user identifiable information here. If they want a random ID, just generate one 🤷

Reply:
There's no user-identifiable information left after hashing with the salt; also see discussion on the ticket on alternatives like location-info.

Reply:
if you adding a random int, you might as well just generate a uuid.

Reply:
@VolkerStolz , what is this "instance URL" you are using? If it is the "external access" URL from the network configuration settings, is it really safe to assume that it has been configured? I haven't configured it on my installation, because the only outside access I currently have is by using an SSH tunnel to a bastion host on my network.